### PR TITLE
GUI test toggling between indicators

### PR DESCRIPTION
**Pull request by emre2038**
## Description
GUI test for https://github.com/OpenUpSA/wazimap-ng-ui/pull/469
Merged https://github.com/OpenUpSA/wazimap-ng-ui/pull/445 into this 

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally
- [ ] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out
- [ ] commit messages are meaningful

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

**Comment by emre2038** [comment]
fixed the e2e test

File: __tests__/gui/toggling_indicators.feature [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Feature: Facility Modal
```
Comment (adieyal) [inline __tests__/gui/toggling_indicators.feature:1]: @emre2038 I don't think this title fits

File: __tests__/gui/toggling_indicators.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+  Scenario: Verify the print view is correct
+    Given I am on the SANEF Homepage
+      Then I wait until map is ready
+    And I expand Data Mapper
+      Then Data Mapper should be displayed
+    And I click on Demographics in Data Mapper
+      Then I select an indicator
+    And I select another indicator
+      Then I check if everything is zero
```
Comment (adieyal) [inline __tests__/gui/toggling_indicators.feature:11]: Wouldn't it be better to call this  `Then I check that everything is not zero`?

**Comment by adieyal** [comment]
The gui tests are failing -

yarn test:gui                                                                                               1 ✘  56s   .env  
yarn run v1.22.11
$ start-server-and-test start http://localhost:1234 cypress:gui
1: starting server using command "npm run start"
and when url "[ 'http://localhost:1234' ]" is responding with HTTP status code 200
running tests using command "npm run cypress:gui"


> wazimap-ng@1.0.0 start
> parcel ./src/index.html


> wazimap-ng@1.0.0 cypress:gui
> cypress run --browser chrome --spec __tests__/gui/*.feature


⚠ Warning: It looks like you're passing --spec a space-separated list of arguments:

"__tests__/gui/facility_modal.feature __tests__/gui/toggling_indicators.feature __tests__/gui/tutorial.feature"

This will work, but it's not recommended.

If you are trying to pass multiple arguments, separate them with commas instead:
  cypress run --spec arg1,arg2,arg3
  
The most common cause of this warning is using an unescaped glob pattern. If you are
trying to pass a glob pattern, escape it using quotes:
  cypress run --spec "**/*.spec.js"

Server running at http://localhost:45295 - configured port 1234 could not be used.
⠸ Building path.js...
====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    8.3.0                                                                              │
  │ Browser:    Chrome 92 (headless)                                                               │
  │ Specs:      3 found (gui/facility_modal.feature, gui/toggling_indicators.feature, gui/tutorial │
  │             .feature)                                                                          │
  │ Searched:   __tests__/gui/facility_modal.feature, __tests__/gui/toggling_indicators.feature, _ │
  │             _tests__/gui/tutorial.feature                                                      │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  gui/facility_modal.feature                                                      (1 of 3)
✨  Built in 3.75s.


  Facility Modal
    1) Verify the print view is correct


  0 passing (12s)
  1 failing

  1) Facility Modal
       Verify the print view is correct:
     AssertionError: Timed out retrying after 4000ms: expected '[ <div.point-mapper__h1_content.default-state--closed>, 1 more... ]' to be 'visible'

This element `[ <div.point-mapper__h1_content.default-state--closed>, 1 more... ]` is not visible because it has CSS property: `display: none`
      at Context.eval (http://localhost:1234/__cypress/tests?p=__tests__/gui/facility_modal.feature:113:99)
      at Context.resolveAndRunStepDefinition (http://localhost:1234/__cypress/tests?p=__tests__/gui/facility_modal.feature:10853:29)
      at Context.eval (http://localhost:1234/__cypress/tests?p=__tests__/gui/facility_modal.feature:10198:35)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     11 seconds                                                                       │
  │ Spec Ran:     gui/facility_modal.feature                                                       │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /home/adie/Development/wazimap-ng-ui/cypress/screenshots/gui/facility_modal.feat     (1280x720)
     ure/Facility Modal -- Verify the print view is correct (failed).png                            


  (Video)

  -  Started processing:  Compressing to 32 CRF                                                     
  -  Finished processing: /home/adie/Development/wazimap-ng-ui/cypress/videos/gui/fac    (0 seconds)
                          ility_modal.feature.mp4                                                   


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  gui/toggling_indicators.feature                                                 (2 of 3)


  Facility Modal
    1) Verify the print view is correct


  0 passing (9s)
  1 failing

  1) Facility Modal
       Verify the print view is correct:
     CypressError: Timed out retrying after 4050ms: `cy.click()` failed because this element is not visible:

`<div class="truncate">Populat...</div>`

This element `<div.truncate>` is not visible because its parent `<div.data-category__h1_content>` has CSS property: `display: none`

Fix this problem, or use `{force: true}` to disable error checking.

https://on.cypress.io/element-cannot-be-interacted-with
      at $Cy.ensureVisibility (http://localhost:1234/__cypress/runner/cypress_runner.js:148473:24)
      at runAllChecks (http://localhost:1234/__cypress/runner/cypress_runner.js:136145:14)
      at retryActionability (http://localhost:1234/__cypress/runner/cypress_runner.js:136199:16)
      at tryCatcher (http://localhost:1234/__cypress/runner/cypress_runner.js:13212:23)
      at Function.Promise.attempt.Promise.try (http://localhost:1234/__cypress/runner/cypress_runner.js:10486:29)
      at tryFn (http://localhost:1234/__cypress/runner/cypress_runner.js:153280:21)
      at whenStable (http://localhost:1234/__cypress/runner/cypress_runner.js:153318:12)
      at http://localhost:1234/__cypress/runner/cypress_runner.js:152810:16
      at tryCatcher (http://localhost:1234/__cypress/runner/cypress_runner.js:13212:23)
      at Promise._settlePromiseFromHandler (http://localhost:1234/__cypress/runner/cypress_runner.js:11147:31)
      at Promise._settlePromise (http://localhost:1234/__cypress/runner/cypress_runner.js:11204:18)
      at Promise._settlePromise0 (http://localhost:1234/__cypress/runner/cypress_runner.js:11249:10)
      at Promise._settlePromises (http://localhost:1234/__cypress/runner/cypress_runner.js:11329:18)
      at Promise._fulfill (http://localhost:1234/__cypress/runner/cypress_runner.js:11273:18)
      at http://localhost:1234/__cypress/runner/cypress_runner.js:12887:46
  From Your Spec Code:
      at Context.eval (http://localhost:1234/__cypress/tests?p=__tests__/gui/toggling_indicators.feature:420:87)
      at Context.resolveAndRunStepDefinition (http://localhost:1234/__cypress/tests?p=__tests__/gui/toggling_indicators.feature:10852:29)
      at Context.eval (http://localhost:1234/__cypress/tests?p=__tests__/gui/toggling_indicators.feature:10197:35)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     8 seconds                                                                        │
  │ Spec Ran:     gui/toggling_indicators.feature                                                  │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /home/adie/Development/wazimap-ng-ui/cypress/screenshots/gui/toggling_indicators     (1280x720)
     .feature/Facility Modal -- Verify the print view is correct (failed).png                       


  (Video)

  -  Started processing:  Compressing to 32 CRF                                                     
  -  Finished processing: /home/adie/Development/wazimap-ng-ui/cypress/videos/gui/tog    (0 seconds)
                          gling_indicators.feature.mp4                                              


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  gui/tutorial.feature                                                            (3 of 3)


  Tutorial (?)
    ✓ Verify the sandbox information is displayed (7382ms)


  1 passing (7s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     7 seconds                                                                        │
  │ Spec Ran:     gui/tutorial.feature                                                             │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF                                                     
  -  Finished processing: /home/adie/Development/wazimap-ng-ui/cypress/videos/gui/tut    (0 seconds)
                          orial.feature.mp4                                                         


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  gui/facility_modal.feature               00:11        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  gui/toggling_indicators.feature          00:08        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  gui/tutorial.feature                     00:07        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  2 of 3 failed (67%)                      00:27        3        1        2        -        -  

Error: Command failed with exit code 2: npm run cypress:gui
    at makeError (/home/adie/Development/wazimap-ng-ui/node_modules/start-server-and-test/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/home/adie/Development/wazimap-ng-ui/node_modules/start-server-and-test/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 2: npm run cypress:gui',
  command: 'npm run cypress:gui',
  escapedCommand: '"npm run cypress:gui"',
  exitCode: 2,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by adieyal** [comment]
@emre2038 I merged 484 - could you please resolve conflicts in this PR so that I can review and merge?

File: __tests__/gui/toggling_indicators.feature (hunk 1 above)
Comment (emre2038) [reply __tests__/gui/toggling_indicators.feature]: done

File: __tests__/gui/toggling_indicators.feature (hunk 2 above)
Comment (emre2038) [reply __tests__/gui/toggling_indicators.feature]: updated the step description

**Comment by emre2038** [comment]
resolved the conflicts @adieyal

**Comment by emre2038** [comment]
closing this since we already have the test coverage merged into staging & master and this PR is obsolete now

**Pull request closed without merging**